File: js-pr/56/javascript/code-quality-readability-maintainability/api_client_37.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
async function makeHttpRequest(url, method = 'GET', data = null, headers = null) {
    const axios = require('axios');
    
    const requestConfig = { headers };
    if (data) {
        requestConfig.data = data;
    }
    
    try {
        const response = await executeRequest(method, url, requestConfig);
        return response.status === 200 ? response.data : null;
    } catch (error) {
        console.error('HTTP request failed:', error.message);
        return null;
    }
}

function executeRequest(method, url, config) {
    const methodHandlers = {
        'GET': () => axios.get(url, config),
        'POST': () => axios.post(url, config.data, config),
        'PUT': () => axios.put(url, config.data, config),
        'DELETE': () => axios.delete(url, config)
    };
    
    const handler = methodHandlers[method.toUpperCase()];
    return handler ? handler() : Promise.reject(new Error('Unsupported method'));
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/api_client_37.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def make_http_request(url, method='GET', data=None, headers=None):
    import requests
    
    method_handlers = {
        'GET': lambda: requests.get(url, headers=headers),
        'POST': lambda: requests.post(url, json=data, headers=headers),
        'PUT': lambda: requests.put(url, json=data, headers=headers),
        'DELETE': lambda: requests.delete(url, headers=headers)
    }
    
    if method not in method_handlers:
        return None
    
    try:
        response = method_handlers[method]()
        return response.json() if response.status_code == 200 else None
    except requests.RequestException:
        return None
# {/fact}